### PR TITLE
DAOS-7312 Test: Functional testing for EC auto object class selection

### PR DESCRIPTION
**Pull request by omar-ocampo**
Fixed highest_epoch() function.
    Added 2 more servers, for a total of 3.
    Created pool uses all servers.
    Replaced SX object class for EC_2P1GX.

Quick-build: true
Skip-unit-tests: true
Skip-nlt: true
Skip-unit-test: true
Skip-func-test-vm: true
Skip-func-hw-test-small: true
Skip-func-hw-test-large: true
Test-tag-hw-medium: pr ioaggregation

Signed-off-by: Omar Ocampo <omar.ocampo.coronado@intel.com>

### CODE REVIEW

File: src/tests/ftest/io/io_aggregation.yaml [hunk 1]
```diff
@@ -16,19 +18,19 @@ server_config:
...
   createset:
     setname: daos_server
   createsize:
-    scm_size: 5000000000
-    nvme_size: 10000000000
+    size: 45000000000
   createsvc:
     svcn: 1
   control_method: dmg
+  nranks: 3
```
Comment (ravalsam) [inline src/tests/ftest/io/io_aggregation.yaml:33]: This test should work without nranks too

File: src/tests/ftest/io/io_aggregation.yaml [hunk 2]
```diff
@@ -16,19 +18,19 @@ server_config:
...
 pool:
   createmode:
     mode_RW:
       mode: 146
   createset:
     setname: daos_server
   createsize:
-    scm_size: 5000000000
-    nvme_size: 10000000000
+    size: 45000000000
```
Comment (ravalsam) [inline src/tests/ftest/io/io_aggregation.yaml:29]: Better to use pool % so it can work on smaller cluster

Comment (omar-ocampo) [reply src/tests/ftest/io/io_aggregation.yaml:29]: Used in dev environment:
scm_size: 20%
nvme_size: 50%

worked fine

File: src/tests/ftest/io/io_aggregation.yaml (hunk 1 above)
Comment (omar-ocampo) [reply src/tests/ftest/io/io_aggregation.yaml]: Should be resolved

File: src/tests/ftest/io/io_aggregation.yaml (hunk 2 above)
Comment (omar-ocampo) [reply src/tests/ftest/io/io_aggregation.yaml]: Should be resolved

**Comment by omar-ocampo** [comment]
@ravalsam Tests within io/ using SX have been changed to EC object like. 
There are other several test folders to be updated, but I do believe is healthier to split them into different PRs.

**Comment by ravalsam** [comment]
> @ravalsam Tests within io/ using SX have been changed to EC object like.
> There are other several test folders to be updated, but I do believe is healthier to split them into different PRs.

Sure, make sense. Some time test might need to be debug and want to try few things so let's keep pushing the stable PR.

**Comment by ravalsam** [comment]
- dfuse_caching_check.py test ran in Ci?
- I do not find the io/io_aggregation.py test result in Ci but you have the tag. Did it run part of Ci?
- io/parallel_io.py has the failure but it looks like the generic test issue as it has the same failure in the Weekly run https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing/206/artifact/Functional%20Hardware%20Medium/io/parallel_io.py/job.log
- io/seg_count and io/large_file_count.py did not run because of some other cleanup error. Can you try to run again in Ci?

**Comment by omar-ocampo** [comment]
@ravalsam, tests are failing due to:
```
Error
The ./io/seg_count.py test was skipped due to host communication error attempting to clean out leftover logs from a previous test run prior to running this test
Standard Output
See launch.py command output for more details
```
It is running one more time, the issue seems to be un-related to this change.
If changes are ok, make sure to approve them and suggest other people who can review this.

**Comment by omar-ocampo** [comment]
Fixed merge problems and triggered re-run

**Comment by omar-ocampo** [comment]
> * dfuse_caching_check.py test ran in Ci?
> * I do not find the io/io_aggregation.py test result in Ci but you have the tag. Did it run part of Ci?
> * io/parallel_io.py has the failure but it looks like the generic test issue as it has the same failure in the Weekly run https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing/206/artifact/Functional%20Hardware%20Medium/io/parallel_io.py/job.log
> * io/seg_count and io/large_file_count.py did not run because of some other cleanup error. Can you try to run again in Ci?

Still getting cleanup errors, other tests seems to be still running, will check in the morning

**Comment by omar-ocampo** [comment]
> * dfuse_caching_check.py test ran in Ci?
> * I do not find the io/io_aggregation.py test result in Ci but you have the tag. Did it run part of Ci?
> * io/parallel_io.py has the failure but it looks like the generic test issue as it has the same failure in the Weekly run https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing/206/artifact/Functional%20Hardware%20Medium/io/parallel_io.py/job.log
> * io/seg_count and io/large_file_count.py did not run because of some other cleanup error. Can you try to run again in Ci?

dfusecachingcheck is not running due to incorrect hw pool, it is on small
ioaggregation is on hw  small, will modify pragma
parallelio is working now
segcount and large_file_count.py keep failing, any suggestions?

**Review by saurabhtandan — COMMENTED** [review]
Changes LGTM, but the tests are failing in CI due to limp.so issue going on in CI currently. Also, as seg_count.py test is not run in CI right now, so you don't need to run it as part of this PR.

**Pull request closed without merging**